mongodbatlas/resource_mongodbatlas_cluster.go

```diff
@@ -1175,7 +1178,19 @@ func expandReplicationSpecs(d *schema.ResourceData) ([]matlas.ReplicationSpec, e
 					id = oldID
 				}
 			}
-			regionsConfig, err := expandRegionsConfig(spec["regions_config"].(*schema.Set).List())
+
+			replaceRegion := ""
+			originalRegion := ""
+
+			if okPRName && d.Get("provider_name").(string) == "GCP" && cast.ToString(d.Get("cluster_type")) == "REPLICASET" {
```

Comment:
[q] I am not sure why we have the check only for `REPLICASET`

Reply:
For my understanding, since the user created a basic cluster, in response it creates automatically with cluster type `REPLICASET` along the replication specs , this why I added validation, for this special scenario